### PR TITLE
Add Azure Backend into Dev

### PR DESCRIPTION
**Pull request by shahidee44**
**##** Description

- Added `DeviceAzure` into devices.py
- Added Notebook 12, testing the azure backend within OpenQAOA.
- Update Qiskit QPU Backend's error message to account for different input Device objects
- Update setup.py
- Fixed Logging Bug for Azure Backends

## Checklist

[//]: <> (- [ ] My code follows the style guidelines of this project)

- [x] I have performed a self-review of my code.
- [x] I have commented my code and used numpy-style docstrings
- [x] I have made corresponding updates to the documentation.
- [x] My changes generate no new warnings
- [x] I have added/updated tests to make sure bugfix/feature works.
- [x] New and existing unit tests pass locally with my changes.

[//]: <> (- [ ] Any dependent changes have been merged and published in downstream modules)

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## How Has This Been Tested?

Name the new unit-tests that you have added along with this change.

12_testing_notebook.ipynb

### CODE REVIEW

File: tests/credentials.json [hunk 1]
```diff
@@ -5,11 +5,9 @@
      "GROUP": "IBMQ_GROUP", 
      "PROJECT": "IBMQ_PROJECT"
  }, 
- "AWS":
+ "AZURE":
```
Comment (Q-lds) [inline tests/credentials.json:8]: Do we still need this file?

File: tests/test_qpu_devices.py [hunk 2]
```diff
@@ -290,6 +291,99 @@ def test_check_connection_provider_wrong_backend_provided_credentials(self):
...
+    available QPUs can be established.
+
+    For any tests using provided credentials, the tests will only pass if those
+    details provided are correct/valid with Azure.
+    """
+    
+    @pytest.mark.api
+    def setUp(self):
+
+        try:
```
Comment (Q-lds) [inline tests/test_qpu_devices.py:308]: Why are we testing things with the credentials file?

Comment (shahidee44) [reply tests/test_qpu_devices.py]: Removed

File: docs/source/index.rst [hunk 3]
```diff
@@ -241,6 +241,7 @@ Contents
    notebooks/09_RQAOA_example.ipynb
    notebooks/10_workflows_on_Amazon_braket.ipynb
    notebooks/11_Mixer_example.ipynb
+   notebooks/12_testing_azure.ipynb
```
Comment (vishal-ph) [inline docs/source/index.rst:244]: should testing_azure be an example notebook?

Comment (Q-lds) [reply docs/source/index.rst:244]: No, probably not. But we will need to revamp all the notebooks anyway. So we can keep it there for the moment (provided that is re-run and the warnings are removed)

**Comment by shahidee44** [comment]
The last 2 commits solved a couple of bugs: 

1. DeviceAzure was missing the `n_qubits` argument which caused it to crash with the QiskitBackend. (The newer version of the backend has a n_qubits check, the object was updated for consistency)

2. The ugly warning messages from using the Azure Backend is now gone. For the most part, the culprit is in one of our dependencies that imported a file deep within the pyquil library that set the logging level to DEBUG. In order to resolve this, the `devices.py` file now sets the logging level back to `ERROR`. Since the Device objects are quiticential to OQ, as long as it is imported and used in whatever workflow the user is doing, the logging is set to the appropriate level.

File: tests/credentials.json [hunk 4]
```diff
@@ -4,12 +4,5 @@
      "HUB": "IBMQ_HUB", 
      "GROUP": "IBMQ_GROUP", 
      "PROJECT": "IBMQ_PROJECT"
- },
```
Comment (vishal-ph) [inline tests/credentials.json:7]: Why was this removed? Don't we need it for GitHub testing?

Comment (vishal-ph) [reply tests/credentials.json:7]: Is it because we are reading the credentials already configured on the machine we are running the tests (i.e. `q128`)?

**Review by vishal-ph — APPROVED** [review]
